draft-ietf-tls-esni.md

```diff
@@ -388,23 +440,34 @@ ClientHelloOuter, and a digest of the complete ClientHelloInner.
 
 When sending ClientHello, the client first computes ClientHelloInner, including
 any PSK binders, and then MAY substitute extensions which it knows will be
-duplicated in ClientHelloOuter. To do so, the client computes a hash H of the
+duplicated in ClientHelloOuter. To do so, the client computes a hash of the
 entire ClientHelloInner message with the same hash as for the KDF used to
-encrypt ClientHelloInner. Then, the client removes and and replaces extensions
-from ClientHelloInner with a single "outer_extensions" extension. The list of
-outer_extensions include those which were removed from ClientHelloInner, in the
-order in which they were removed. The hash contains the full ClientHelloInner
-hash H computed above.
-
-This process is reversed by client-facing servers upon receipt. Specifically,
-the server replaces the "outer_extensions" with extensions contained in
-ClientHelloOuter. The server then computes a hash H' of the reconstructed
-ClientHelloInner. If H' does not equal OuterExtensions.hash, the server aborts
-the connection with an "illegal_parameter" alert.
+encrypt ClientHelloInner. [[TODO: Will there always be a hash function
```

Comment:
Yes -- each KDF has an associated hash function.

Reply:
From the spec, it seems like all that's required of the KDF is that it's an "extract-then-expand" KDF:
https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hpke-05#section-4

In fact, the currently assigned code points are all for HKDF:
https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hpke-05#section-7.2

But AFAICT there's no requirement that the KDF always have some underlying hash function. As an alternative to HKDF, one can imagine a dedicated primitive that satisfies the extract-then-expand API but is not constructed from a hash function.

Sorry if this is nit-picky :)

Reply:
I see your point. Let's replace this with `Extract` instead of `Hash` and call it a day.

Reply:
Well, that would change the protocol. Better to make it an issue?

Reply:
Yep!

Reply:
Done. https://github.com/tlswg/draft-ietf-tls-esni/issues/261